### DISCUSSION TITLE
Add permission creative check

### DISCUSSION THREAD

**Discussion by Guest1131** [EssentialsX/Essentials#4232] [Q&A] [answered]
Hi, I noticed that if I have the permission:

essentials.gamemode.creative
essentials.gamemode

On the group for example Founder, I can use /gamemode 1, and so far so good, everything works. But if I change the group instead of Founder I put Default, where I don't have gamemode permissions. The creative remains, and I can use it even if I have the group without permissions.

Could a check be added? If I don't have permissions, I am automatically put in survival gamemode.

**Comment by pop4959** [comment, marked as answer]
If you don't want someone to have access to creative game mode then do not give them permission. If you remove their permission you will have to run commands to put them in the "correct" game mode (be that survival, or whatever else on your server).

**Reply by Guest1131** [reply]
You're missing my point. I'm talking about the fact that if I change groups, I still have creative without me using the /gamemode command

**Reply by mdcfe** [reply]
`/gamemode` is a command to change your gamemode, and the `essentials.gamemode.<gamemode>` permissions control the game odes available **through the command**. 

EssentialsX does not track permissions changes to enforce gamemode changes. Doing so would add a significant amount of burden and complexity (polling permissions, tracking who set what gamemode, determining whether or not other plugins could affect that tracking etc.), and it's unlikely EssentialsX would be able to cover most sensible use cases. 

If someone changes their gamemode and you subsequently remove access to the command, you'll need to ensure they're put back into the correct gamemode.

**Reply by Guest1131** [reply]
"If someone changes their gamemode and you subsequently remove access to the command, you'll need to ensure they're put back into the correct gamemode."

And how can I do this?
